### PR TITLE
423 remove service discovery

### PR DESCRIPTION
**Pull request by jacksondelametter**
PR to remove the Service Discovery module. This also adds to the Draft Release Notes that the module has been removed as well as fix a commit hash reference to the removal of Sagemaker.

### CODE REVIEW

File: DRAFT_RELEASE_NOTES.md [hunk 1]
```diff
@@ -101,4 +101,5 @@ If you added any custom configurations to the `sparkoperator` service account pe
 
 # What's Changed
 - `pyproject.toml` files updated to allow for Python version `>=3.8`.
-- All SageMaker modules and corresponding references were removed in commit bdcbb409.
+- All SageMaker modules and corresponding references were removed in commit 8ce3f39.
```
Comment (carter-cundiff) [inline DRAFT_RELEASE_NOTES.md:104]:
```suggestion
- All SageMaker modules and corresponding references were removed in commit 8ce393f.
```

File: DRAFT_RELEASE_NOTES.md [hunk 2]
```diff
@@ -101,4 +101,5 @@ If you added any custom configurations to the `sparkoperator` service account pe
...
+- The Service Discovery module and corresponding references were removed in commit 4a1d7abf.
```
Comment (carter-cundiff) [inline DRAFT_RELEASE_NOTES.md:105]: You currently have 2 commits, so this is probably gonna change if you squash them

Comment (jacksondelametter) [reply DRAFT_RELEASE_NOTES.md:105]: Unfortunately I need to have two commits to do this. When you squash them, git changes the commit hash.

File: DRAFT_RELEASE_NOTES.md (hunk 1 above)
Comment (jacksondelametter) [reply DRAFT_RELEASE_NOTES.md]: Fixed

File: DRAFT_RELEASE_NOTES.md (hunk 2 above)
Comment (carter-cundiff) [reply DRAFT_RELEASE_NOTES.md]: Oh that makes sense, can't predict the squash commit hash

**Review by carter-cundiff — APPROVED** [review]

**Pull request merged**